File: src/commands/reset.ts
import chalk from "chalk";
import { clearState, clearAuth } from "../config/store";

export async function resetCommand(options: { auth?: boolean }): Promise<void> {
	await clearState();
	console.log(chalk.green("  Sync state cleared. Next sync will re-download all documents."));

	if (options.auth) {
		await clearAuth();
		console.log(chalk.green("  Device disconnected. Run `remarkable-cli init` to reconnect."));
	}

	console.log();
}

File: src/commands/reset.rs
use crate::config::ConfigStore;
use crate::error::Result;
use colored::Colorize;

pub async fn run(store: &ConfigStore, auth: bool) -> Result<()> {
    store.clear_state().await?;
    println!("{}", "Sync state cleared.".green().bold());
    println!("The next sync will re-download all documents.");

    if auth {
        store.clear_auth().await?;
        println!("{}", "Device authentication removed.".green().bold());
        println!("Run `remarkable init` to reconnect.");
    }

    println!();
    Ok(())
}
